sheerwater_benchmarking/forecasts/gefs.py

```diff
@@ -0,0 +1,505 @@
+            'longitude': 'lon',
+            'ensemble_member': 'member'
+        }
+    )
+
+    # Handle different run types
+    if run_type == 'average':
+        # Take ensemble mean
+        ds = ds.mean(dim='member')
+    elif isinstance(run_type, int):
+        # Take specific ensemble member
+        ds = ds.isel(member=run_type)
+    elif run_type != 'perturbed':
+        raise ValueError(f"Invalid run_type: {run_type}")
+
+    # TODO: Select the right variable?
+    # Should that be done here caching each variable separately
+    # or in the gefs function caching all variables together?
+
+    # TODO: is forecast_type a relevant argument for this dataset?
+
+    # TODO: is time_group a relevant argument for this dataset?
+
+    # TODO: figure out what field conversions/aggregations need to be done
+    # TODO: figure out what fields can be dropped
+    return ds
+
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['variable', 'forecast_type',
+                       'run_type', 'time_group', 'grid'],
+           cache=False,
+           timeseries=['start_date', 'model_issuance_date'],
+           chunking={"lat": 300, "lon": 300, "lead_time": 1,
+                     "start_date": 1000,
+                     "model_issuance_date": 1000, "start_year": 1,
+                     "member": 1})
+def gefs_gridded(start_time, end_time, variable, forecast_type,
+                run_type='average', time_group='weekly', grid="global1_5"):
+    """GEFS forecast data with regridding."""
+    ds = gefs_raw(start_time, end_time, variable, forecast_type,
+                 run_type, time_group=time_group, grid='global0_25')
+
+    # TODO: verify if this is necessary
+    # Convert to base180 longitude
+    ds = lon_base_change(ds, to_base="base180")
+
+    if grid == 'global0_25':
+        return ds
+
+    # Regrid onto appropriate grid
+    ds = regrid(ds, grid, base='base180', method='conservative')
+    return ds
+
+
+@dask_remote
+@cacheable(data_type='array',
+           timeseries='time',
+           cache=False,
+           cache_args=['variable', 'lead', 'prob_type', 'grid', 'mask', 'region'])
+def gefs(start_time, end_time, variable, lead, prob_type='deterministic',
+        grid='global1_5', mask='lsm', region="global"):
+    """Standard format forecast data for GEFS forecasts.
+
+    Args:
+        start_time: Start time for the forecast period
+        end_time: End time for the forecast period
+        variable: Variable to forecast (tmp2m, precip, etc.)
+        lead: Forecast lead time (week1, week2, etc.)
+        prob_type: Type of probability forecast (deterministic or ensemble)
+        grid: Grid resolution (global0_25, global1_5)
+        mask: Mask to apply (lsm, None)
+        region: Region to clip to (global, africa, conus)
+    """
+    lead_params = {
+        "week1": ('weekly', 0),
+        "week2": ('weekly', 7),
+        "week3": ('weekly', 14),
+        "week4": ('weekly', 21),
```

Comment:
I think we can get a week5 in here

Reply:
I added week5, and removed the weekly lead_time variable as part of removing lead_time from a previous comment. I am unclear on if I should remove the weeks12, weeks23, weeks34 leads now because without the lead_time var I don't know how those leads work? Should I leave them or remove them?